src/lib/eina/eina_file_win32.c

```diff
@@ -917,24 +917,35 @@ eina_file_map_new(Eina_File *file, Eina_File_Populate rule,
    if (!map)
      {
         HANDLE fm;
+        SYSTEM_INFO sys_info;
+
+        GetSystemInfo(&sys_info);
+
         map = malloc(sizeof (Eina_File_Map));
         if (!map)
           {
              eina_lock_release(&file->lock);
              return NULL;
           }
 
+        const DWORD align = ~(sys_info.dwAllocationGranularity - 1);
```

Comment:
Perhaps it would be best to include some comments explaining the logic behind these operations. I have a vague notion of what's being done and I'm having trouble understanding this. It might seem even more obscure to future readers.